src/runtime_src/hip/core/module.cpp

```diff
@@ -24,6 +26,15 @@ module_xclbin(std::shared_ptr<context> ctx, const std::string& file_name)
   create_hw_context();
 }
 
+module_xclbin::
+module_xclbin(std::shared_ptr<context> ctx, void* data, size_t size)
+  : module{std::move(ctx), true}
+{
+  std::vector<char> buf(static_cast<char*>(data), static_cast<char*>(data) + size);
+  m_xrt_xclbin = xrt::xclbin{buf};
+  create_hw_context();
```

Comment:
Prefer initializer list, avoid ctor body.
Initialize the data members in the order needed, something like this:
```
: module(...)
, m_xrt_xclbin{create_xclbin(data, size)}
, m_hw_context{get_context()->get_xrt_device, m_xrt_xclbin.get_uuid()}
{}
```
where create_xclbin() is a static free namespaced function.